packages/techdocs-common/src/stages/publish/local.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import fetch from 'cross-fetch';
+import express from 'express';
+import fs from 'fs-extra';
+import { Logger } from 'winston';
+import { Entity, EntityName } from '@backstage/catalog-model';
+import {
+  resolvePackagePath,
+  PluginEndpointDiscovery,
+} from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import { PublisherBase, PublishRequest } from './types';
+
+/* `remoteUrl` is the URL which serves files from the local publisher's static directory. */
+export type LocalPublishReturn = Promise<{ remoteUrl: string }>;
+
+const staticDocsDir = resolvePackagePath(
```

Comment:
Is there a reason we stage files locally within the plugin directory itself, instead of say, a directory within [os.tmpdir()](https://nodejs.org/api/os.html#os_os_tmpdir)?

I realize local file system is not the recommended way to deploy TechDocs, but I would imagine some cloud hosting providers restrict write access to code paths for security reasons.

Reply:
Maybe @hooloovooo has more context on why we chose the `static` directory in the plugin root for this purpose. TechDocs uses a temporary directory in its prepare and generate steps. But publishes in a more non-temp location. Perhaps to make sure the published files stay for longer duration and are not removed on restarts?

In my opinion, the "working directory" should be configurable (similar to Scaffolder), so that users can specify a suitable path for the job. https://github.com/backstage/backstage/issues/3333